### PR TITLE
Initial work to add azuread_device table

### PR DESCRIPTION
**Pull request by chrichts**
# Example query results
<details>
  <summary>Results</summary>
  
```
Add example SQL query results here (please include the input queries as well)
```
</details>

### CODE REVIEW

**Comment by chrichts** [comment]
@cbruno10 - looks like the linter is failing because its running out of memory. I ran it locally and had no issues.

**Comment by cbruno10** [comment]
Hey @chrichts , yes unfortunately it's a known issue based on the size of the MS Graph Go SDK, please see https://github.com/microsoftgraph/msgraph-sdk-go/issues/129 for more information.

If it's passing locally, can you please paste your terminal output in this PR? Thanks!

**Comment by chrichts** [comment]
Cool, looks like it passed this time around. I'll create a separate PR for the other issues I mentioned in Slack,

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@chrichts Thanks for submitting this PR! The table code looks great, I've added a few minor suggestions.

File: azuread/table_azuread_device.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+		List: &plugin.ListConfig{
+			Hydrate: listDevices,
+			KeyColumns: plugin.KeyColumnSlice{
+				// Key fields
+				{Name: "display_name", Require: plugin.Optional},
+			},
+		},
+
+		Columns: []*plugin.Column{
+
```
[the remark this answers is not in the document]
Comment (chrichts) [reply azuread/table_azuread_device.go:37]: These are marked in the API as "For internal use only". Not 100% sure what that means, but my experience is that properties marked as such don't contain any meaningful data.

https://docs.microsoft.com/en-us/graph/api/resources/device?view=graph-rest-1.0#properties

[the remark this answers is not in the document]
Comment (chrichts) [reply azuread/table_azuread_device.go:37]: Let me know if you still want me to add this anyway

**Comment by Subhajit97** [comment]
Hi @chrichts, thanks for the quick updates!
One more suggestion related to [MSGraph Go SDK](https://github.com/microsoftgraph/msgraph-sdk-go) version update.

As per https://github.com/microsoftgraph/msgraph-sdk-go/issues/129#issuecomment-1235546314, there are some build improvements in `v0.37.0`, and we have updated our plugin recently to use  [MSGraph Go SDK](https://github.com/microsoftgraph/msgraph-sdk-go) `v0.37.0`.

Can you please update your branch and made the required changes as per the latest SDK? Thanks!

**Comment by chrichts** [comment]
@Subhajit97 - updated the go sdk version. Let me know if there is anything else

**Review by Subhajit97 — APPROVED** [review]

**Comment by Subhajit97** [comment]
Thanks again, @chrichts, for this addition! We'll release this in our next plugin release.

**Pull request merged**